### PR TITLE
added basic auth

### PR DESCRIPTION
**Pull request by mohammadkhizerkhan**
image
image

### CODE REVIEW

File: my-app/src/components/productpage/productlisting/Product.jsx [hunk 1]
```diff
@@ -35,8 +54,8 @@ function Product({item}) {
         </div>
       </div>
       <div class="card-btn-container">
-        <button class="btn product-btn btn-s">Add To Wishlist</button>
-        <button class="btn product-btn btn-s">Add To Cart</button>
+        <button class="btn product-btn btn-s" onClick={()=>addWishlistHandler(token,item)}>Add To Wishlist</button>
+        <button class="btn product-btn btn-s" onClick={()=>addCartHandler(token,item)}>Add To Cart</button>
```
Comment (Ash-D23) [inline my-app/src/components/productpage/productlisting/Product.jsx:48]: there is no need to pass token here

File: my-app/src/containers/home/Navbar.jsx [hunk 2]
```diff
@@ -16,9 +18,8 @@ function Navbar() {
         <ul class="nav-ul">
           <li>
             <Link
-              to="wishlist"
+              to={token?"wishlist":"login"}
```
Comment (Ash-D23) [inline my-app/src/containers/home/Navbar.jsx:21]: you can add a variable path above and if token is true set it to /wishlist

**Review by Ash-D23 — APPROVED** [review]

File: my-app/src/backend/db/users.js [hunk 3]
```diff
@@ -16,4 +16,27 @@ export const users = [
...
+
+// {
+//   "email": "testGameCart@gmail.com",
+//      "password": "test123"
+//   }
+
+
+// {
+//   "firstName": "John",
+//   "lastName": "Doe",
```
Comment (rutvikpumak) [inline my-app/src/backend/db/users.js:39]: remove this

File: my-app/src/components/auth/SignUp.jsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+            signup(form)
+        }
+        setform({
+            firstname:"",
+            lastname:"",
+            email:"",
+            password:""
+        })
+    }
+
```
Comment (rutvikpumak) [inline my-app/src/components/auth/SignUp.jsx:32]: remove this

File: my-app/src/components/productpage/productlisting/Product.jsx [hunk 5]
```diff
@@ -11,6 +14,22 @@ function Product({item}) {
...
   } = item;
+  const addCartHandler=(token,item)=>{
+    if(token){
+      navigate("/carts")
+    }
+    else{
+      navigate("/login")
+    }
+  }
+  const addWishlistHandler=(token,item)=>{
```
Comment (rutvikpumak) [inline my-app/src/components/productpage/productlisting/Product.jsx:20]: use ternary

File: my-app/src/store/data/AuthContext.jsx [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    const localToken=JSON.parse(localStorage.getItem("login"))
+    const localUser=JSON.parse(localStorage.getItem("user"))
+    const [token,setToken]=useState(localToken?.token)
+    const [user,setUser]=useState(localUser?.user)
+    const [error,setError]=useState("");
+
+    const login=async (email,password)=>{
+        if(email && password !== ""){
+            try {
+                const response=await loginService(email,password);
```
Comment (rutvikpumak) [inline my-app/src/store/data/AuthContext.jsx:18]: remove consoles

File: my-app/src/store/reducer/reducer.js [hunk 7]
```diff
@@ -22,4 +22,15 @@ const ProductReducer = (state, action) => {
   }
 };
 
+// const AuthReducer=(state,action)=>{
+//   switch (action.type) {
```
Comment (rutvikpumak) [inline my-app/src/store/reducer/reducer.js:26]: remove consoles

**Review by rutvikpumak — COMMENTED** [review]
Remove all the consoles and make suggested changes

File: my-app/src/store/reducer/reducer.js (hunk 7 above)
Comment (mohammadkhizerkhan) [reply my-app/src/store/reducer/reducer.js:26]: done

File: my-app/src/components/productpage/productlisting/Product.jsx (hunk 5 above)
Comment (mohammadkhizerkhan) [reply my-app/src/components/productpage/productlisting/Product.jsx:20]: done

**Review by rutvikpumak — APPROVED** [review]

**Pull request merged**